### DISCUSSION TITLE
【讨论】为何作者不鼓励使用 COBOL 商业语言编程？

### DISCUSSION THREAD

**Discussion by yikeke** [yikeke/tao-of-programming#1] [Q&A] [answered]
COBOL 科普：其名称来自于通用商业语言（英语：Common Business Oriented Language，又译为面向常规业务型语言/常规商业信息处理语言）的缩写，是最早的高阶编程语言、以及是世界上最早实施标准化的计算机语言之一。属于编译语言。（[维基百科](https://zh.wikipedia.org/wiki/COBOL)）

求解：为什么作者在 1987 年写成该书的时候，就不鼓励使用 COBOL 商业语言来编程呢？

**Comment by cliuxinxin** [comment]
要么没有体现阴阳，要么就是无法在TAO中找到位置

**Comment by sanjotsubaki** [comment, marked as answer]
COBOL语言写的程序里满是大写字母，代码极长，最恐怖的是它们是粘起来的，以下是我找到的一个小段过程部代码：
MULTIPLYQUANTITYBYPRICEGIVING
//MULTIPLY(乘)QUANTITY(数量)BY(被......)PRICE(单价)GIVING（给出）
大概意思是把用户给出的数量乘以单价。
听说这种语言在日本还有很多银行里的设备使用，所以有国内的程序员能拿到一些外包。

**Reply by huiscliu** [reply]
对变量与操作符都要两边加空格的, 喜欢加空行的, 密集恐怖证患者, 这种程序就是谋杀....

**Comment by fyzzy1943** [comment]
这是时代的河流，作出的选择；旧时代的技艺，应该回到旧时代的记忆里

**Comment by willshion** [comment]
太恐怖了
MULTIPLYQUANTITYBYPRICEGIVING
//MULTIPLY(乘)QUANTITY(数量)BY(被......)PRICE(单价)GIVING（给出）
文言文还要 自己 切

**Comment by moocstudent** [comment]
打卡

**Comment by cliuxinxin** [comment]
moocstudent <notifications@github.com> 于2020年12月6日周日 下午5:28写道：

**Reply by Henrik-Xu** [reply]
我是旧时代的残党，新时代没有能载我的船

**Reply by wolfangy** [reply]
以前在投资银行里面工作，整个这边的部门有且只有一个同事在做这个，他在那个岗位上工作了三十年。。。不知道算不算悟道了
